builder/processor/src/main/java/io/helidon/builder/processor/FactoryMethods.java

```diff
@@ -0,0 +1,327 @@
+
+        // factory method
+        String createMethod = "create";
+
+        List<TypeInfo> candidates = configObjectCandidates.stream()
+                .map(it -> processingContext.toTypeInfo(it))
+                .flatMap(Optional::stream)
+                .toList();
+
+        for (TypeInfo typeInfo : candidates) {
+            // is this a config object?
+            if (doesImplement(typeInfo, PROTOTYPE_TYPE)) {
+                // it should have create(Config) with the correct typing
+                Optional<FactoryMethod> foundMethod = typeInfo
+                        .elementInfo()
+                        .stream()
+                        .filter(TypeInfoPredicates::isMethod)
+                        .filter(TypeInfoPredicates::isStatic)
+                        .filter(it -> createMethod.equals(it.elementName()))
+                        .filter(it -> it.parameterArguments().size() == 1)
+                        .filter(it -> CONFIG_TYPE.equals(it.parameterArguments().get(0).typeName()))
+                        .filter(it -> typeInfo.typeName().equals(it.typeName()))
+                        .findFirst()
+                        .map(it -> new FactoryMethod(typeInfo.typeName(), typeInfo.typeName(), createMethod, CONFIG_TYPE));
+                if (foundMethod.isPresent()) {
+                    return foundMethod;
+                }
+            }
+        }
+
+        for (TypeInfo typeInfo : candidates) {
+            // if the target type implements ConfiguredType, we use the generic parameter of that interface to look for our config
+            // look for "implements ConfiguredType"
+
+            if (doesImplement(typeInfo, RUNTIME_OBJECT_TYPE)) {
+                // there is no config factory method available for the type that we have
+                TypeName candidateTypeName = typeInfo.typeName();
+                // we are now interested in a method with signature "static T create(Config)" where T is the type we are handling
+                Optional<FactoryMethod> foundMethod = typeInfo
+                        .elementInfo()
+                        .stream()
+                        .filter(TypeInfoPredicates::isMethod)
+                        .filter(TypeInfoPredicates::isStatic)
+                        .filter(it -> createMethod.equals(it.elementName()))
+                        .filter(it -> it.parameterArguments().size() == 1)
+                        .filter(it -> CONFIG_TYPE.equals(it.parameterArguments().get(0).typeName()))
+                        .filter(it -> candidateTypeName.equals(it.typeName()))
+                        .findFirst()
+                        .map(it -> new FactoryMethod(candidateTypeName, candidateTypeName, createMethod, CONFIG_TYPE));
+                if (foundMethod.isPresent()) {
+                    return foundMethod;
+                }
+            }
+        }
+
+        // TODO when in the same module, we need to use different approach through definition
```

Comment:
same here

Reply:
replaced with comment that explains the problem and kind of "fingers crossed" - it never failed during my changes to Helidon.